(maint) Add supercronic binary to container for k8s

underscorgan opened the pull request:
Please add all notable changes to the "Unreleased" section of the CHANGELOG.

This supersedes #1043 and includes some general cleanup and brings the r10k dockerfiles/automation up to date with other projects

Xtigyro committed:
Add Supercronic for Use in K8s

Signed-off-by: Miroslav Hadzhiev <miroslav.hadzhiev@gmail.com>

Commit:
(maint) Add supercronic to the release Dockerfile, and a little cleanup

Commit:
(maint) Move changes from 8a744719042e3fb046fc99061effa05edb989833 to the release Dockerfile

Commit:
(maint) Update github workflow to publish release containers

Commit:
(maint) Add in missing hadolint ignores

Iristyle commented on docker/r10k/Dockerfile:
```diff
@@ -32,24 +37,29 @@ COPY docker/r10k/docker-entrypoint.d /docker-entrypoint.d
     gem install --no-doc /r10k.gem && \
-    rm -f /r10k.gem
+    rm -f /r10k.gem && \
+    curl --fail --silent --show-error --location --remote-name "$supercronic_url" && \
```
A bit of this I would do differently for buildkit caching (i.e. use `ADD` to pull the release file)... but I need to put up a PR to merge together the 2 Dockerfiles with stages, so I'll address it there. Should be able to get rid of the `pipefail` at the same time as well)

Commit:
(maint) Use docker-login action

Iristyle commented on .github/workflows/docker.yml:
```diff
@@ -10,13 +10,18 @@ jobs:
     steps:
       - uses: actions/checkout@master
+      - uses: azure/docker-login@v1
+        with: # This doesn't seem to work unless we point directly to the secrets
```
Yeah, I don't see anything special in the code - https://github.com/Azure/docker-login/blob/master/src/login.ts

mwaggett commented on .github/workflows/docker.yml:
```diff
@@ -28,5 +33,24 @@ jobs:
         run: |
-          docker login -u "$DOCKERHUB_USERNAME" -p "$DOCKERHUB_PASSWORD"
           make publish
+      - name: Build release container
```
Will these get triggered the same way as the dev/non-release builds?

underscorgan replied:
yep, this also runs on every commit, but just realized this container is lacking the logic to only build a new release container if one hasn't already been published for the latest version

mwaggett replied:
okay cool, yeah, I was wondering what was preventing it from building release containers for dev versions

Commit:
(maint) Add logic to makefile for conditional release builds

Note: This also includes a move to `git describe --tags` instead of `git
describe`. This is due to the tag action creating lightweight tags. The
other tag action I found that supports lightweight tags has a whole host
of other issues so this feels like the best solution.
